pubsub/topic.go

```diff
@@ -316,6 +337,15 @@ func (t *Topic) updateRequest(cfg TopicConfigToUpdate) *pb.UpdateTopicRequest {
 		pt.MessageStoragePolicy = messageStoragePolicyToProto(cfg.MessageStoragePolicy)
 		paths = append(paths, "message_storage_policy")
 	}
+	if cfg.RetentionDuration != 0 {
+		if cfg.RetentionDuration > 0 {
+			pt.MessageRetentionDuration = durationpb.New(cfg.RetentionDuration)
+		} else {
+			// If set to a negative value, clear MessageRetentionDuration.
+			pt.MessageRetentionDuration = nil
+		}
```

Comment:
Nit, total matter of preference: invert the if/else to make the positive-value case the "green path" and the negative-value case a special case (thus indented).
```suggestion
              pt.MessageRetentionDuration = durationpb.New(cfg.RetentionDuration)
		if cfg.RetentionDuration < 0 {
			// If set to a negative value, clear MessageRetentionDuration.
			pt.MessageRetentionDuration = nil
		}
```

Reply:
Done.